camunda/camunda-platform-helm
# Add identity and keycloak to charts

- [x] Add identity as sub-chart
  - [x] Add identity manifest templates
  - [x] Add potential variables to values file (+ doc)
  - [x] Add keycloak as dependency
    - [x] Set/overwrite necessary values
    - [x] Document values
  - [x] Connect identity with keycloak
     - [x] use service names to connect with keycloak
     - [x] use keycloak generated secrets
     - [x] allow to overwrite secrets -which will also change configuration in identity
    - [x] golden files for defaults
    - [x] property test for conditions, e.g. existing secret
  - [x] Adjust CI
    - [x] adjust makefile to get the deps for the sub chart

--------------------

**Breakthrough:**

Manually tested via:

```sh
helm repo add bitnami https://charts.bitnami.com/bitnami
helm dependency update charts/ccsm-helm/charts/identity/
helm install zell-helm-test charts/ccsm-helm/ --set zeebe.enabled=false --set operate.enabled=false --set tasklist.enabled=false --set elasticsearch.enabled=false --set identity.image.tag=SNAPSHOT
k port-forward svc/zell-helm-test-identity 8080:80
k port-forward svc/zell-helm-test-keycl 18080:80 # without that we were not able to see the identity login page?
```

![identity](https://user-images.githubusercontent.com/2758593/160018745-567dc39c-4c1c-4501-b4c2-28e7bf2b17e6.png)


Added multiple new template tests. Integration tests verifies that pods become ready at least, more to come.

related #127
